src/xclim/indices/_conversion.py

```diff
@@ -1319,6 +1319,58 @@ def _get_D_from_M(time):  # noqa: N802
     )
 
 
+# TODO: How can I check exact units for inputs
+# @declare_units(
+#     net_radiation="[MJ m-2 day-1]",
+#     tas="[degC]",
+#     wind="[m s-1]",
+#     es="[kPa]",
+#     ea="[kPa]",
+#     delta_svp="[kPa degC-1]",
+#     gamma="[kPa degC]",
+#     G="[MJ m-2 day-1]",
+# )
+def fao_allen98(net_radiation, tas, wind, es, ea, delta_svp, gamma, G=0):
+    r"""
+    FAO-56 Penman-Monteith equation.
+
+    Estimates reference evapotranspiration (ETo) from a hypothetical short grass reference surface (
+    height = 0.12m, surface resistance = 70 s m-1, albedo  = 0.23 and a ``moderately dry soil surface resulting from
+    about a weekly irrigation frequency``).
+    Based on equation 6 in based on :cite:t:`allen_crop_1998`.
+
+    Parameters
+    ----------
+    net_radiation : xarray.DataArray
+        Net radiation at crop surface [MJ m-2 day-1].
+    tas : xarray.DataArray
+        Air temperature at 2 m height [degC].
+    wind : xarray.DataArray
+        Wind speed at 2 m height [m s-1].
+    es : xarray.DataArray
+        Saturation vapour pressure [kPa].
+    ea : xarray.DataArray
+        Actual vapour pressure [kPa].
+    delta_svp : xarray.DataArray
+        Slope of saturation vapour pressure curve [kPa degC-1].
+    gamma : xarray.DataArray
+        Psychrometric constant [kPa deg C].
+    G : float, optional
+        Soil heat flux (G) [MJ m-2 day-1] (For daily default to 0).
+
+    Returns
+    -------
+    xarray.DataArray
+        Potential Evapotranspiration from a hypothetical grass reference surface [mm day-1].
+    """
+    tasK = convert_units_to(tas, "K")
+    a1 = 0.408 * delta_svp * (net_radiation - G)
+    a2 = gamma * 900 / (tasK) * wind * (es - ea)
+    a3 = delta_svp + (gamma * (1 + 0.34 * wind))
+
+    return (a1 + a2) / a3
```

Comment:
```suggestion
    tasK = convert_units_to(tas, "K")
    wind = convert_units_to(wind, "m s-1")
    es= convert_units_to(es,  "kPa")
    ea= convert_units_to(ea, "kPa")
    delta_svp= convert_units_to(delta_svp, "kPa degC-1")
    gamma= convert_units_to(gamma, "kPa degC")
    G= convert_units_to(G, "MJ m-2 day-1")
    a1 = 0.408 * delta_svp * (net_radiation - G)
    a2 = gamma * 900 / (tas) * wind * (es - ea)
    a3 = delta_svp + (gamma * (1 + 0.34 * wind))

    return (a1 + a2) / a3
```

Reply:
@aulemahal  is this case extreme enough to allow a list comprehension in xclim? :3

```python
    net_radiation,tas,wind,es,ea,delta_svp, gamma, G = [convert_units_to(da,u) 
    for da, u in zip(
        [net_radiation,tas,wind,es,ea,delta_svp, gamma, G],
        "MJ m-2 day-1", "K", "m s-1", "kPa", "kPa", "kPa degC-1", "kPa degC", "MJ m-2 day-1"
    )
```

Reply:
Not too bad. I still prefer the long form, but this is on the limit of readable.